dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -228,6 +233,23 @@ private Host findByNameNotDefault(final String siteName, final User user, final
         return null;
     }
 
+    @Override
+    public Optional<Host> findByIdOrKey(final String siteIdOrKey, final User user,
```

Comment:
Let's included an Integration Test for this method

Reply:
Done!